### PR TITLE
[Admin] - Gab/approve application integration

### PR DESCRIPTION
**Pull request by gidoo5331**

### CODE REVIEW

File: apps/admin/modules/creator-application/approve/index.tsx [hunk 1]
```diff
@@ -27,23 +28,23 @@ export const ApproveApplicationModal = ({
...
-  const { mutate, status } = useApproveApplication();
+  const { mutateAsync, isPending: isLoading } = useApproveCreatorApplication();
 
-  const handleSubmit = () => {
-    mutate(data!.id);
-    if (status == 200) {
-      setOpened(false);
+  const handleSubmit = async () => {
+    try {
+      await mutateAsync(data!.id)
```
Comment (arkhurst) [inline apps/admin/modules/creator-application/approve/index.tsx:35]: if data can be undefined, i think we should handle that error
maybe:

`
  if(!data?.id) {
    // handle error
  }
`

File: apps/admin/modules/creator-application/approve/index.tsx [hunk 2]
```diff
@@ -61,8 +62,8 @@ export const ApproveApplicationModal = ({
           </DialogDescription>
         </DialogHeader>
         <DialogFooter>
-          <Button type="button" onClick={() => handleSubmit()}>
-            Yes, Approve
+          <Button disabled={isLoading} type="button" onClick={() => handleSubmit()}>
+          { isLoading && (<Loader2Icon className="animate-spin" />)} Yes, Approve
```
Comment (arkhurst) [inline apps/admin/modules/creator-application/approve/index.tsx:66]: None blocking: But I normally prefer

`{ isLoading ? <Loader2Icon className="animate-spin" /> : null }`

**Review by arkhurst — CHANGES_REQUESTED** [review]

File: apps/admin/modules/creator-application/reject/index.tsx [hunk 3]
```diff
@@ -27,25 +26,25 @@ export const RejectApplicationModal = ({
...
+  //       duration: 5000,
+  //     });
+  //   } else {
+  //     toast({
+  //       title: "Error rejecting application",
+  //       variant: "destructive",
+  //       duration: 5000,
+  //     });
+  //   }
+  // };
```
Comment (Bendomey) [inline apps/admin/modules/creator-application/reject/index.tsx:47]: why is this commented out?

**Review by Bendomey — CHANGES_REQUESTED** [review]

File: apps/admin/modules/creator-application/reject/index.tsx (hunk 3 above)
Comment (gidoo5331) [reply apps/admin/modules/creator-application/reject/index.tsx:47]: Will do integration for reject in another PR, and I deleted the testData I was using before you gave me the Api.

**Review by Bendomey — APPROVED** [review]

**Pull request merged**